contracts/retirement/RetireToucanCarbon.sol

```diff
@@ -185,6 +185,10 @@ contract RetireToucanCarbon is
                 address(this)
             );
 
+            // The smallest denomination that can be retired is a kilo,
+            // anything less will fail.
+            if (balance < 1e15) continue;
+
```

Comment:
Obviously, the side effect here is that for "dust" or micro-amounts intended or not, no retirement is possible, and that the user keeps the TCO2 in their wallet. The reasoning was an easier to audit Certificate NFT and that micro offsetting might cause more CO2 emission than saved.